File: JAVA/Bubble Sort/BubbleSort.java
public class BubbleSort {
    public static void main (String[] args) {
        
        int array[] = {5,8,2,7,25,14,1,65,3};
        
        System.out.println("Original Array : ");
        for(int num : array )
        System.out.print(num+"\t");  
                  
        bubbleSort(array);  

        System.out.println("\nSorted Array : ");
        for(int num : array )
        System.out.print(num+"\t");  
        
    }
    
    public static void bubbleSort(int arr[]) {
        for(int i=0; i < arr.length; i++){  
            for(int j=1; j < (arr.length - i); j++){ 
                int temp;
                if(arr[j-1] > arr[j]){   
                       temp = arr[j-1];  
                       arr[j-1] = arr[j];  
                       arr[j] = temp;  
                }  
            }  
        }
    } 
}

File: Go/Bubble Sort/bubble_sort.go
/*Copyright 2020 Denis Lituev
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
    http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

// Code for Bubble Sort in Go
package main

import "fmt"

// BubbleSort perfoms bubble sort in ascending mode
func BubbleSort(arr []int) {
	size := len(arr)
	for i := 0; i < (size - 1); i++ {
		for j := 0; j < size-i-1; j++ {
			if arr[j] > arr[j+1] {
				/* Swapping */
				arr[j+1], arr[j] = arr[j], arr[j+1]
			}
		}
	}
}

func main() {
	data := []int{9, 1, 8, 2, 7, 3, 6, 4, 5}
	BubbleSort(data)
	fmt.Println(data)
}
